File: app/src/main/java/leavesc/hello/retrofit2_rxjava2/MainActivity.java
package leavesc.hello.retrofit2_rxjava2;

import android.arch.lifecycle.ViewModel;
import android.os.Bundle;
import android.view.View;

import leavesc.hello.retrofit2_rxjava2.view.BaseActivity;


public class MainActivity extends BaseActivity {

    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    protected ViewModel initViewModel() {
        return null;
    }

    public void queryWeather(View view) {
        startActivity(QueryWeatherActivity.class);
    }

    public void queryIDCard(View view) {
        startActivity(IDCardActivity.class);
    }

    public void createQrCode(View view) {
        startActivity(QrCodeActivity.class);
    }
}

File: app/src/main/java/leavesc/hello/weather/ui/MainActivity.kt
package leavesc.hello.weather.ui

import android.os.Bundle
import android.text.TextUtils
import leavesc.hello.weather.R
import leavesc.hello.weather.core.cache.AreaCache
import leavesc.hello.weather.core.view.BaseActivity

/**
 * 作者：leavesC
 * 时间：2019/5/31 15:39
 * 描述：
 */
class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (TextUtils.isEmpty(AreaCache.getAdCode(this))) {
            startActivity(MapActivity::class.java)
        } else {
            startActivity(WeatherActivity::class.java)
        }
        finish()
    }

}